engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/nodevalidation/NodeDataValidator.scala

```diff
@@ -73,16 +77,53 @@ class CustomNodeValidator(modelData: ModelData) extends NodeDataValidator[Custom
 
   private val nodeValidator = new GenericNodeTransformationValidator(expressionCompiler, expressionEvaluator)
 
+  protected def validate(transform: SingleInputGenericNodeTransformation[_],
+                         nodeData: NodeData with WithParameters,
+                         validationContext: ValidationContext, outputVar: Option[String])(implicit metaData: MetaData): ValidationPerformed = {
+    implicit val nodeId: NodeId = NodeId(nodeData.id)
+    nodeValidator.validateNode(transform, nodeData.parameters, validationContext, outputVar) match {
+      case Valid(result) =>
+        ValidationPerformed(result.errors, Some(result.parameters))
+      case Invalid(e) => ValidationPerformed(e.toList, None)
+    }
+  }
+
+}
+
+class CustomNodeValidator(val modelData: ModelData) extends WithParametersNodeValidator[CustomNode] {
+
   override def validate(nodeData: CustomNode, validationContext: ValidationContext)(implicit metaData: MetaData): ValidationResponse = {
     val transformer = modelData.processWithObjectsDefinition.customStreamTransformers(nodeData.nodeType)._1.obj
     transformer match {
```

Comment:
Thise fragment we can extract to some mixin and reuse it in SinkNodeValidator and SourceNodeValidator.

Reply:
I extracted a bit